src/shared-styles/_space.scss

```diff
@@ -0,0 +1,106 @@
+@use "baseline";
+@use "error";
+@use "logical";
+@use "media-query";
+@use "rem";
+@use "size";
+@use "validate";
+
+@function calculate($block-size, $unit-size: baseline.for-units(4)) {
+  $remaining: $unit-size - $block-size;
+
+  @if ($remaining > 0) {
+    @return $remaining;
+  }
+
+  @return $unit-size;
+}
+
+// Fallbacks for CSS logical properties contained within this mixin require the following treatment of HTML dir attributes:
+//  - document level: always specified, via the HTML element
+//  - block level: specified on every element within a block describing a direction switch.
+//
+// For example:
+// <html lang="..." dir="ltr">
+// ...
+// <div>
+//   Doesn't need a dir attribute. Most cases will be like this.
+// </div>
+//
+//<div class="test" dir="rtl">
+//
+//  This block changes the text direction. Every descendant element must have its own dir attribute....
+//
+//  <div class="test" dir="rtl">... even if the direction doesn't change.</div>
+//
+//  <div class="test" dir="ltr">But obviously also when it does.</div>
+//
+//</div>
+@mixin _spacing($sizes, $space-type, $dimension: "") {
```

Comment:
Can this go at the bottom of the file? Can't remember if ordering is important, but public things should come first when possible.

Reply:
Agree. Will move.

You can generally ignore execution order when the file only contains mixins / functions. It's when you get those units included in the same file they're defined in that order starts to matter.